graylog2-server/src/main/java/org/graylog2/rest/resources/system/DeflectorResource.java

```diff
@@ -58,33 +53,32 @@
 
     private final IndexSetRegistry indexSetRegistry;
     private final ActivityWriter activityWriter;
-    private final Map<String, Provider<RotationStrategy>> rotationStrategies;
-    private final ClusterConfigService clusterConfigService;
-    private final ElasticsearchConfiguration configuration;
 
     @Inject
     public DeflectorResource(IndexSetRegistry indexSetRegistry,
-                             ActivityWriter activityWriter,
-                             Map<String, Provider<RotationStrategy>> rotationStrategies,
-                             ClusterConfigService clusterConfigService,
-                             ElasticsearchConfiguration configuration) {
+                             ActivityWriter activityWriter) {
         this.indexSetRegistry = indexSetRegistry;
         this.activityWriter = activityWriter;
-        this.rotationStrategies = rotationStrategies;
-        this.clusterConfigService = clusterConfigService;
-        this.configuration = configuration;
     }
 
     @GET
     @Timed
     @ApiOperation(value = "Get current deflector status")
     @RequiresPermissions(RestPermissions.DEFLECTOR_READ)
     @Produces(MediaType.APPLICATION_JSON)
-    public DeflectorSummary deflector() throws TooManyAliasesException {
-        // TODO 2.2: Resource needs to be adjusted to support multiple write targets
-        final IndexSet indexSet = indexSetRegistry.getAllIndexSets().stream()
-                .findFirst()
-                .orElseThrow(() -> new NotFoundException("Unable to find current deflector alias"));
+    @Deprecated
+    public DeflectorSummary deprecatedDeflector() throws TooManyAliasesException {
+        throw new UnsupportedOperationException("Unsupported since Graylog 2.2 - use /system/deflector/{indexSetId}");
```

Comment:
I introduced a default marker for index sets and changed the resources to use that. :+1: